packages/shared/getComponentName.js

```diff
@@ -43,27 +43,28 @@ function getWrappedName(
   );
 }
 
-function getComponentName(type: mixed): string | null {
-  if (type == null) {
+function getComponentName(elementType: mixed): string | null {
+  if (elementType == null) {
     // Host root, text node or just invalid type.
     return null;
   }
   if (__DEV__) {
-    if (typeof (type: any).tag === 'number') {
+    // TODO: protect against potential false positives from user-defined statics
+    if (typeof (elementType: any).tag === 'number') {
```

Comment:
The check was introduced in #13197. Seems like this was meant as a helper for the transition period since it was used on the full fiber previously.

It's probably better to [remove this altogether](https://github.com/eps1lon/react/commit/2be613fc1cb9f286ec434e1692820ea325fba9c3). An unknown component name should appear in the tests leading to a fail instead of a different error message.

[Using `_debugID`](https://github.com/eps1lon/react/commit/4403e4bfb40efc7ca7446144c64bd3d7f5d02cfb) is likely best to reduce the number of false positives.

[Using `isValidElementType`](https://github.com/eps1lon/react/commit/a2737cdb2f59ca033b2aa37ee2f1e1bee608c40d) leads to breakage in the [internal React.lazy tests](https://github.com/facebook/react/blob/9055e31e5c82d03f0a365c459f7bc79e402dbef5/packages/react-reconciler/src/__tests__/ReactLazy-test.internal.js#L562).